azurerm/resource_arm_sql_virtual_network_rule.go

```diff
@@ -0,0 +1,141 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/sql/mgmt/2015-05-01-preview/sql"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmSqlVirtualNetworkRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmSqlVirtualNetworkRuleCreateUpdate,
+		Read:   resourceArmSqlVirtualNetworkRuleRead,
+		Update: resourceArmSqlVirtualNetworkRuleCreateUpdate,
+		Delete: resourceArmSqlVirtualNetworkRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				//TODO: Validation (invalid if non-numeric characters)
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"server_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"virtual_network_subnet_id": {
```

Comment:
🤔 do you think it'd be worth making this `subnet_id` to be more consistent with the other Terraform resources?

Reply:
@tombuildsstuff Yes, that definitely makes sense. 

On a similar note, what is your opinion on `ignore_missing_vnet_service_endpoint`? Do you think it would be better if it was shorted a bit as well to `ignore_missing_service_endpoint` or even `ignore_missing_endpoint`?

Reply:
given the description ([in the API Docs](https://docs.microsoft.com/en-us/rest/api/sql/virtualnetworkrules/createorupdate)): `Create firewall rule before the virtual network has vnet service endpoint enabled.` - I guess leaving it as `ignore_missing_vnet_service_endpoint ` would probably make the most sense? Whilst I don't feel that name's particularly descriptive, it's how the Azure API's chosen to label it, so it probably makes sense to match it incase it does extra stuff in the future

Reply:
✅